src/signin.js

```diff
@@ -1,101 +1,133 @@
       emailInput.style.borderColor = '#CCD5E3'
     }
   }
-}
+})
 
 //비밀번호 에러 메세지 출력
-function passErrorMessage (e) {
-  //에러메세지 - 값이 없을 경우
-  if(passInput.value.trim() === '') {
-    errorMessage(passInput, '비밀번호를 입력해주세요.');
+const passInput = document.querySelector('#pass');
+passInput.addEventListener('focusout', (e) => {
+  const passwordEmpty = passInput.value.trim() === '';
+  const findClassNameErrorMessage = passInput.parentElement.querySelector('.errorMessage');
+  
+  if(passwordEmpty) {
+    setErrorMessageToElement (passInput, '비밀번호를 입력해주세요.');
   }
   //올바른 값으로 수정
   else if (validatePass(passInput.value)) {
-    if (passInput.parentElement.parentElement.querySelector('.errorMessage')) {
-      passInput.parentElement.parentElement.querySelector('.errorMessage').remove();
+    if (findClassNameErrorMessage) {
+      findClassNameErrorMessage.remove();
       passInput.style.borderColor = '#CCD5E3'
     }
   }
-}
-
-//로그인 시도
-function submitForm (e) {
-  //현재페이지에서 테스트
-  e.preventDefault();
-
-  if (emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101') {
-    window.location.href = "./folder";
-  } else {
-    errorMessage(emailInput, '이메일을 확인해주세요.');
-    errorMessage(passInput, '비밀번호를 확인해주세요.');
-  }
-}
+});
 
 //눈 모양 클릭
-function toggleIcon (e) {
-  if (toggle.previousElementSibling.type === 'password') {
-    toggle.previousElementSibling.type = 'text';
+const toggle = document.querySelector('.fa-solid');
+toggle.addEventListener('click', (e) => {
+  const toggleIcon = toggle.previousElementSibling;
+  
+  if (toggleIcon.type === 'password') {
+    toggleIcon.type = 'text';
     toggle.classList.remove('fa-eye-slash');
     toggle.classList.add('fa-eye');
   } else {
-    toggle.previousElementSibling.type = 'password';
+    toggleIcon.type = 'password';
     toggle.classList.remove('fa-eye');
     toggle.classList.add('fa-eye-slash');
   }
+});
+
+//로그인 시도
+// signForm.addEventListener('submit', (e) => {
+  //   //현재페이지에서 테스트
+  //   e.preventDefault();
+  
+  //   const tryCorrectId = emailInput.value === 'test@codeit.com' && passInput.value === 'codeit101'
+  
+  //   if (tryCorrectId) {
+    //     window.location.href = "./folder";
+    //   } else {
+      //     setErrorMessageToElement (emailInput, '이메일을 확인해주세요.');
+//     setErrorMessageToElement (passInput, '비밀번호를 확인해주세요.');
+//   }
+// });
+
+const signForm = document.querySelector('.signin');
+signForm.addEventListener('submit', async (e) => {
+  e.preventDefault();
+  const userEmail = emailInput.value;
+  const userPassword = passInput.value;
+
+  try {
+    const response = await fetch ('https://bootcamp-api.codeit.kr/docs/api/sign-in', {
+      method: 'POST', 
+      headers: {
+        'Content-Type': 'application/json',
+      }, 
+      body: JSON.stringify({ userEmail, userPassword }),
+    });
+    
+    if (response.ok) {
+      const result = await response.json();
+      localStorage.setItem('accessToken', result.accessToken);
```

Comment:
타입스크립트와 다르게 자바스크립트는 어떤 데이터 형식을 가지고 있는지 모르기때문에, 그런것이 명확하지 않으면 항상 `console.log`로 내가 어떤 값을 받게 되는지 로그를 찍어보는 습관을 가지면 좋습니다.
타입스크립트에도 마찬가지입니다. 

`await response.json()`으로 리턴되는 데이터는 result가 아니라 result.data에 있습니다. 
`result.data`안에 `accessToken`이 있습니다. 

```suggestion
      const result = await response.json();
      localStorage.setItem('accessToken', result.data.accessToken);
```